feed-formats/buttwoo-v1.js

```diff
@@ -92,14 +98,25 @@ module.exports = function init(ssb) {
       //   format: 'buttwoo-v1',
       //   data,
       // })
-      feedFormat._msgIdCache.set(nativeMsg, msgId)
+      feedFormat._msgIdStringCache.set(nativeMsg, msgId)
       return msgId
     },
 
     _getMsgIdBFE(nativeMsg) {
-      const [encodedValue, signature] = feedFormat._extract(nativeMsg)
-      const data = blake3.hash(Buffer.concat([encodedValue, signature]))
-      return Buffer.concat([BUTTWOO_MSG_TF, data])
+      if (feedFormat._msgIdBFECache.has(nativeMsg)) {
+        return feedFormat._msgIdBFECache.get(nativeMsg)
+      }
+
+      let data = feedFormat._msgIdCache.get(nativeMsg)
+      if (!data) {
+        const [encodedValue, signature] = feedFormat._extract(nativeMsg)
+        data = blake3.hash(Buffer.concat([encodedValue, signature]))
+        feedFormat._msgIdCache.set(nativeMsg, data)
+      }
```

Comment:
Nice optimization! Perhaps these 5 lines above can be refactored into a `_getMsgIdBuf` method that both `getMsgId` and `_getMsgIdBFE` can make use of.